SummerProject/middleware/user_auth.py

```diff
@@ -0,0 +1,19 @@
+# pylint: disable=R0205,R1710,R0201
+
+"""Custom middleware"""
+from django.http.response import HttpResponseBadRequest
+
+
+class AuthenticatedUser(object):
+    """Middleware to verify that the user is authenticated"""
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        return self.get_response(request)
+
+    def process_request(self, request):
+        """Method to verify that the user is authenticated"""
+        if not request.user.is_authenticated and request.path == '/api/v1/user/update_info/':
```

Comment:
`/api/v1/user/update_info/` us not RESTfull URL name. Lets change it to `/api/v1/user/<user_id>/` or `/api/v1/user/self/`?